### PR TITLE
Storybook color and stats automation

### PR DESCRIPTION
**Pull request by daverau-optimizely**
This PR imports a custom-made CSS analysis tool called [csswhat](https://github.com/optimizely/csswhat) into OUI storybook and build toolchain. 

✅ This PR builds in the monolith via `grunt dev` without errors.

1. Adds table of CSS properties for easy reference — remember the OUI 9 docs days? 💃 
2. Adds a list of helpful CSS stats and color lists in Storybook 🎨 
3. Adds new yarn commands, but you don't need that, just run `yarn storybook` as normal 😉 
4. Outputs stats to console for quick reference — this helps quickly answer "did my latest save just add new unique colors to OUI or significantly increase the total package size?" etc. (for example:🤔while rebasing this PR the total unique color count increased from 30 to 40, that's something to investigate — hopefully this tool will help surface those kinds of changes in the future.)

# 1. Table of CSS Properties
Screen Shot 2019-06-07 at 9 52 06 AM

# 2. Storybook Stats and Colors
Screen Shot 2019-06-01 at 8 11 42 PM
Screen Shot 2019-06-01 at 8 11 53 PM

# 3. New Yarn Commands
- `yarn csswhat` — the main command to compile OUI sass into css and run csswhat analytics to output oui.css.json stats (this is run automatically via `yarn build-storybook`)
- `yarn csswhat:watch` — the main command in watch mode (this is run automatically via `yarn storybook`
- `yarn csswhat-css` — compile sass to css only
- `yarn csswhat-json:watch` — same as above + watch
- `yarn csswhat-json` — analyze previously compiled oui.css and output oui.css.json stats
- `yarn csswhat-json:watch` — same as above + watch

Here we remove 3 colors, wait for console, then Storybook updates 👀

colors-csswhat-full

# 4. Command Line Output
```
-------------------
CSSWhat: oui.css
-------------------
Size: 193.3 KB
Rules: 1197
Selectors: 2778
Properties: 154
Declarations: 2396
Values: 557
!important: 317
-------------------
Colors: 51
Frequently used hex colors: 18
#262626 (50), #ffffff (32), #e0e0e0 (27), #080738 (23), #999999 (16), #000000b3 (5), #d9e1ff (5), #b34040 (4), #cc4949 (4), #f7f7f7 (4), #333333 (3), #7391ff (3), #bfbfbf (3), #c7c7c7 (2), #808080 (2), #000000 (2), #f0f0f0 (2), #29a65e (2)
One-of-a-kind hex colors: 30
#26323d, #ddffdd, #ffdddd, #0086b3, #990073, #bb6688, #008080, #000080, #445588, #0000ff, #19469d, #219161, #40a070, #954121, #408080, #f8f8ff, #7b81d1, #888888, #a6a6a6, #ffff00, #ffdede, #d6f5e4, #fff6d6, #ffbdbd, #993737, #d1d1d1, #0027b3, #002ccc, #1a4bff, #5378ff
One-of-a-kind alpha hex colors: 3
#ffffff99, #0000004d, #0000000a
-------------------
```

Some things that could be improved still (for future PRs):
- Search is non-existant — I'd like to pair with someone on some approaches to wire this up — it would be interesting to know how to do it with plain JS and React. In the next few weeks I'd like to get search working for selectors and a handful of other things.
- You'll notice webpack compiles, then csswhat runs, then webpack compiles again — that could be better.
- I'd like to land this and improve incrementally rather than have this sit any longer, but I'm open to big changes if this is really wonky.

### CODE REVIEW

**Comment by daverau-optimizely** [comment]
Hi @lshofmann, FrontendInfra mentioned you unblocked someone who had this issue in the past. Can you help me or point me in the direction to allow the Travis CI process for OUI to access the private csswhat repo? This PR is failing because https://github.com/optimizely/csswhat is a private repo yet still under the Optimizely GitHub org. I thought that would be an automatic allowance.

Thanks for the help!

**Comment by daverau-optimizely** [comment]
No idea why this is failing now as sortable passes for me locally:
```
 PASS  src/components/Sortable/tests/index.js (6.485s)
  components/Sortable
    Normal Sorting
      ✓ should render all draggable items in order (138ms)
      ✓ should allow items to be dragged and reordered (148ms)
      ✓ should render preview of item in drag layer when dragging (108ms)
      ✓ should not reorder if dragging item over its leading target (161ms)
      ✓ should not reorder if dragging item over its trailing target (166ms)
      ✓ should convert ungrouped items to groups when enabling grouping (107ms)
    Grouped Sorting
      ✓ should render all items in groups and in order (52ms)
      ✓ should allow items to be reordered within a group (297ms)
      ✓ should allow items to be dragged outside of a group (378ms)
      ✓ should allow groups to be reordered (486ms)
      ✓ should not reorder if dragging group over itself (394ms)
      ✓ should not reorder if dragging group over another group (642ms)
      ✓ should extract all items from groups when grouping is disabled (73ms)
```

**Comment by daverau-optimizely** [comment]
Turns out this was a yarn.lock issue; checking out yarn.lock from devel and re-running yarn seemed to fix this.

**Comment by ayeshakmaz** [comment]
@daverau-optimizely Have you run the deduplicate command recently? Seeing some duplicate declarations in the yarn.lock that could probably be condensed

File: .storybook/css.story.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+              <td className="border--top soft-double--ends">
+                {css.propertiesObj[key].map((k,index)=>(
+                  <p key={`${i}_${index}`} className="flush push-half--bottom">{k.selector}: <span className="color--brand">{k.value}</span></p>
+                ))}
+              </td>
+            </tr>
+          ))}
+      </table>
+    </div>
+    // <div>
```
Comment (quarklemotion) [inline .storybook/css.story.js]: can we remove this commented out code?

File: package.json [hunk 2]
```diff
@@ -54,6 +54,7 @@
     "babel-plugin-transform-class-properties": "^6.24.1",
     "codecov": "^1.0.1",
     "css-loader": "^0.25.0",
+    "csswhat": "optimizely/csswhat",
```
Comment (quarklemotion) [inline package.json]: I don't have access to this repo, can you add the frontend team as a contributor to this repo?

Comment (daverau-optimizely) [reply package.json]: 👍 Whoops, sorry! I was going to make this a public repo at first and Root mentioned our default preference is to have private repos for tools etc, so I went that route.

Comment (daverau-optimizely) [reply package.json]: Just added all of FE

File: .storybook/css.story.js (hunk 1 above)
Comment (daverau-optimizely) [reply .storybook/css.story.js]: Whoops, yes good catch, thought I removed all that already.

**Comment by daverau-optimizely** [comment]
> Are we OK with breaking the storybook build for all non-optimizely consumers of OUI?

Yeah... so I have a JIRA request open to make the CSSWhat repo public and an open source project here, but that might take a little bit of time so it will break OUI for outside consumers until the csswhat repo is public. Should we be very concerned about that? Maybe making a note is good enough since we have a plan to open source it and fix the issue?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This file is similar to the format suggested by [Keep a CHANGELOG](https://github.com/olivierlacan/keep-a-changelog).
 
 ## Unreleased
+- [Feature] CSSWhat stats about color, selectors, !important, etc. (Breaking change for external consumers of OUI until the csswhat tool is open sourced and the repo is flipped to public by an admin)
```
Comment (ayeshakmaz) [inline CHANGELOG.md]: Nit: might be easier to read if the () note was on a new line and indented

Could you also add the PR link at the end?

**Comment by daverau-optimizely** [comment]
> @daverau-optimizely Have you run the deduplicate command recently? Seeing some duplicate declarations in the yarn.lock that could probably be condensed

Good catch, just ran that and did a new push — 🤞

**Comment by daverau-optimizely** [comment]
> Are we OK with breaking the storybook build for all non-optimizely consumers of OUI?

Personally, yes.

I'm also fine if a GitHub admin just flipped the repo public to avoid the breakage while we wait for legal etc to review my JIRA ticket to open source the project: https://optimizely.atlassian.net/browse/OPEN-95

**Pull request merged**